src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -160,6 +160,27 @@ public bool TryGetLoadedConfig([NotNullWhen(true)] out RuntimeConfig? runtimeCon
         return configLoadSucceeded;
     }
 
+    /// <summary>
+    /// Initialize the runtime configuration provider with the specified accessToken for the specified datasource.
+    /// This initialization method is used to set the access token for the current runtimeConfig.
+    /// As opposed to using a json input and regenerating the runtimconfig, it sets the access token for the current runtimeConfig on the provider.
+    /// </summary>
+    /// <param name="accessToken">The string representation of a managed identity access token</param>
+    /// <param name="dataSourceName">Name of the datasource for which to assign the token.</param>
+    /// <returns>true if the initialization succeeded, false otherwise.</returns>
+    public bool InitializeWithAccesstoken(
+        string? accessToken,
+        string dataSourceName)
+    {
+        if (ManagedIdentityAccessToken.ContainsKey(dataSourceName))
```

Comment:
Did you mean `!ContainsKey()`?

Reply:
This should be containsKey only. it checks if the datasource to which we are trying to assign the accessToken exists in the first place.

Reply:
I see. Thanks for clarifying!